src/transformers/models/swiftformer/modeling_tf_swiftformer.py

```diff
@@ -0,0 +1,892 @@
+
+        res = tf.reshape(x, [-1, height * width, channels])
+        res = self.attn(res)
+        res = tf.reshape(res, [-1, height, width, channels])
+        if self.use_layer_scale:
+            x = x + self.drop_path(self.layer_scale_1 * res, training=training)
+            x = x + self.drop_path(self.layer_scale_2 * self.linear(x), training=training)
+        else:
+            x = x + self.drop_path(res, training=training)
+            x = x + self.drop_path(self.linear(x), training=training)
+        return x
+
+
+class TFSwiftFormerStage(keras.layers.Layer):
+    """
+    A Swiftformer stage consisting of a series of `SwiftFormerConvEncoder` blocks and a final
+    `SwiftFormerEncoderBlock`.
+
+    Input: tensor in shape `[batch_size, channels, height, width]`
+
+    Output: tensor in shape `[batch_size, channels, height, width]`
+    """
+
+    def __init__(self, config: SwiftFormerConfig, index: int, **kwargs) -> None:
+        super().__init__(**kwargs)
+
+        layer_depths = config.depths
+        dim = config.embed_dims[index]
+        depth = layer_depths[index]
+
+        self.blocks = []
+        for block_idx in range(depth):
+            block_dpr = config.drop_path_rate * (block_idx + sum(layer_depths[:index])) / (sum(layer_depths) - 1)
+
+            if depth - block_idx <= 1:
+                self.blocks.append(
+                    TFSwiftFormerEncoderBlock(config, dim=dim, drop_path=block_dpr, name=f"blocks_._{block_idx}")
+                )
+            else:
+                self.blocks.append(TFSwiftFormerConvEncoder(config, dim=dim, name=f"blocks_._{block_idx}"))
+
+    def call(self, input: tf.Tensor, training: bool = False) -> tf.Tensor:
+        for i, block in enumerate(self.blocks):
+            input = block(input, training=training)
+        return input
+
+    def build(self, input_shape=None):
+        for layer in self.blocks:
+            with tf.name_scope(layer.name):
+                layer.build(None)
+
+
+class TFSwiftFormerEncoder(keras.layers.Layer):
+    def __init__(self, config: SwiftFormerConfig, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.config = config
+
+        embed_dims = config.embed_dims
+        downsamples = config.downsamples
+        layer_depths = config.depths
+
+        # Transformer model
+        self.network = []
+        name_i = 0
+        for i in range(len(layer_depths)):
+            stage = TFSwiftFormerStage(config, index=i, name=f"network_._{name_i}")
+            self.network.append(stage)
+            name_i += 1
+            if i >= len(layer_depths) - 1:
+                break
+            if downsamples[i] or embed_dims[i] != embed_dims[i + 1]:
+                # downsampling between two stages
+                self.network.append(TFSwiftFormerEmbeddings(config, index=i, name=f"network_._{name_i}"))
+                name_i += 1
+
+        self.gradient_checkpointing = False
+
+    def call(
```

Comment:
`training` should be added as an arg here and passed to the layers

Reply:
https://github.com/huggingface/transformers/pull/23342/commits/9c03892b4d1cee3c2b6b95a0bd8e0adaf9630f94